src/snapshot/download.js

```diff
@@ -0,0 +1,66 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var toImage = require('../plot_api/to_image');
+var Lib = require('../lib');
+
+/**
+ * @param {object} gd figure Object
+ * @param {object} opts option object
+ * @param opts.format 'jpeg' | 'png' | 'webp' | 'svg'
+ */
+function downloadImage(gd, opts) {
+    
+    // check for undefined opts
+    opts = opts || {};
+    
+    // default to png
+    opts.format = opts.format || 'png';
+    
+    if(Lib.isIE()) {
+        Lib.notifier('Snapshotting is unavailable in Internet Explorer. ' +
```

Comment:
We make this an option.

I think the mode bar `toImage` button should use `Lib.notifier` to display messages (as currently) whereas `Plotly.downloadImage` method should throw `new Error`s.